cloud/cluster/fetch_cron.go

```diff
@@ -0,0 +1,87 @@
+package cluster
+
+import (
+	"time"
+)
+
+type FetchCron struct {
+	tickCh   <-chan time.Time
+	f        Fetcher
+	inCh     chan nodesAndError
+	outCh    chan []NodeUpdate
+	outgoing []NodeUpdate
+	closer   chan struct{}
+	cl       *Cluster
+}
+
+// Returns a full list of visible nodes.
+type Fetcher interface {
+	Fetch() ([]Node, error)
+}
+
+func NewFetchCron(f Fetcher, t time.Duration, cl *Cluster) *FetchCron {
+	c := &FetchCron{
+		tickCh:   time.NewTicker(t).C,
+		f:        f,
+		inCh:     nil,
+		outCh:    make(chan []NodeUpdate),
+		outgoing: nil,
+		closer:   make(chan struct{}),
+		cl:       cl,
+	}
+	go c.loop()
+	return c
+}
+
+func (c *FetchCron) loop() {
+	c.inCh = make(chan nodesAndError)
+	go func() {
```

Comment:
This seems pretty complex.... If Cluster's ignoring us, is it really important to fetch?

Why was the simple thing insufficient?